### PR TITLE
Merge "remove card" after rebase from "list of cards" to master

### LINKED ISSUE #5: [Card Management] Remove card
Requirements
 - define card ID (how to identify which card is being removed)

### PR DESCRIPTION
**Pull request by VaneChira**
branch/functionality "list of cards" was rebased and merged into "Remove card" branch (parent branch). This needs to be merged to master as well.  All the comments from "list of cards" merge request were addressed.

### CODE REVIEW

File: src/main/java/com/bootcamp/demo/controller/CardController.java [hunk 1]
```diff
@@ -1,24 +1,37 @@
...
+import com.google.firebase.cloud.FirestoreClient;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.web.bind.annotation.*;
 
-import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+import java.util.List;
 
 @RestController
-@RequestMapping(value = "/card", produces = APPLICATION_JSON_VALUE )
+@RequestMapping("/card")
```
Comment (remus-lupin-team) [inline src/main/java/com/bootcamp/demo/controller/CardController.java]: `@RequestMapping(value = "/card", produces = APPLICATION_JSON_VALUE )` is the correct annotation for this REST Controller. Please change id back. Please be aware that there might have been errors generated when rebasing.

File: src/main/java/com/bootcamp/demo/controller/CardController.java [hunk 2]
```diff
@@ -1,24 +1,37 @@
...
 public class CardController {
-    private final CardService cardService;
+    private Firestore firestoreDB;
```
Comment (remus-lupin-team) [inline src/main/java/com/bootcamp/demo/controller/CardController.java]: Please use the `FirestoreDao firestoreDao` Bean.

File: src/main/java/com/bootcamp/demo/controller/CardController.java [hunk 3]
```diff
@@ -1,24 +1,37 @@
...
     public CardController(CardService cardService) {
         this.cardService = cardService;
     }
 
-    @DeleteMapping(path="{id}")
-    ResponseEntity<String> removeCard(@PathVariable String id){
-      this.cardService.removeCard(id);
-      return new ResponseEntity<>("Successfully deleted card "+ id, HttpStatus.OK);
+    @DeleteMapping(value = "/{cardNumber}")
+    public String removeCard(@PathVariable String cardNumber){
```
Comment (remus-lupin-team) [inline src/main/java/com/bootcamp/demo/controller/CardController.java]: Please check the previous implementation for removing a card. Using the CardServiceis the correct approach.

File: src/main/java/com/bootcamp/demo/controller/CardController.java (hunk 2 above)
Comment (remus-lupin-team) [reply src/main/java/com/bootcamp/demo/controller/CardController.java]: Please check the previous implementation. Using the CardServiceis the correct approach.

File: src/main/java/com/bootcamp/demo/controller/CardController.java [hunk 4]
```diff
@@ -1,24 +1,37 @@
...
+        this.firestoreDB = FirestoreClient.getFirestore();
+        ApiFuture<WriteResult> writeResult = firestoreDB.collection("cards").document(cardNumber).delete();
+        return "Successfully deleted " + cardNumber;
+    }
+
+    @GetMapping("/getAllCards")
+    @ResponseBody
```
Comment (remus-lupin-team) [inline src/main/java/com/bootcamp/demo/controller/CardController.java]: The `@ResponseBody` annotation is not needed. It is already included in the `@RestController` annotation used for this controller.

File: src/main/java/com/bootcamp/demo/dao/FirestoreDaoImpl.java [hunk 5]
```diff
@@ -1,24 +1,38 @@
...
+import java.util.List;
+import java.util.concurrent.ExecutionException;
 
 import static java.lang.System.getProperty;
 import static java.nio.charset.StandardCharsets.UTF_8;
 
 @Repository
 public class FirestoreDaoImpl implements FirestoreDao{
     private Firestore firestoreDB;
+    DocumentToCardMapper mapper;
```
Comment (remus-lupin-team) [inline src/main/java/com/bootcamp/demo/dao/FirestoreDaoImpl.java]: Please make this field `private final` so it respects the encapsulation principle.

File: src/main/java/com/bootcamp/demo/controller/CardController.java (hunk 1 above)
Comment (VaneChira) [reply src/main/java/com/bootcamp/demo/controller/CardController.java]: Fixed

File: src/main/java/com/bootcamp/demo/controller/CardController.java (hunk 2 above)
Comment (VaneChira) [reply src/main/java/com/bootcamp/demo/controller/CardController.java]: Fixed

File: src/main/java/com/bootcamp/demo/controller/CardController.java (hunk 3 above)
Comment (VaneChira) [reply src/main/java/com/bootcamp/demo/controller/CardController.java]: Fixed

File: src/main/java/com/bootcamp/demo/controller/CardController.java (hunk 4 above)
Comment (VaneChira) [reply src/main/java/com/bootcamp/demo/controller/CardController.java]: Fixed

File: src/main/java/com/bootcamp/demo/dao/FirestoreDaoImpl.java (hunk 5 above)
Comment (VaneChira) [reply src/main/java/com/bootcamp/demo/dao/FirestoreDaoImpl.java]: Fixed

**Review by remus-lupin-team — APPROVED** [review]

**Pull request merged**